### ISSUE TITLE
Cognito userpool tags must be a key/value mapping

### ISSUE THREAD

**Issue by mwgamble** [aws-cloudformation/cfn-lint#2574]
### Is this feature request related to a new rule or cfn-lint capabilities?

rules

### Describe the feature you'd like to request

Most Cloudformation resources, such as Log Groups, accept tags as a list of objects, where each object has two keys (`Key` and `Value`). For example:

```
Tags:
  - Key: my-key
    Value: my-value
```

Cognito userpools don't do this. Instead, they accept tags as a mapping of key/value pairs:

```
UserPoolTags:
  my-key: myvalue
```

I had a deployment fail because this wasn't picked up beforehand, and the documentation doesn't make the format of this clear.

### Describe the solution you'd like

cfn-lint should warn when providing a list for `UserPoolTags` and not an object.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change

**Comment by kftsehk** [comment]
https://github.com/aws-cloudformation/cfn-lint/blob/main/src/cfnlint/data/CloudSpecs/us-east-1.json#L150023-L150028
among other various places in different specs, search "UserPoolTag" for all

According to described, spec should be below, but was `"PrimitiveType": "Json"` which is superset of `List` and `Map`

```
    "UserPoolTags": {
     "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpool.html#cfn-cognito-userpool-userpooltags",
     "PrimitiveItemType": "String",
     "Type": "Map",
     "Required": false,
     "UpdateType": "Mutable"
    },
```

@kddejong replacing spec is a botted process, have to fix this on the bot.

**Comment by kftsehk** [comment]
Funny that not only the spec has an issue, validation of `"Type": "Map"` also had an issue. Both below pass cfn-lint but fail to deploy, note `Lambda.Environment.Variables` and `CognitoUserPool.UserPoolTags` are not `Map`, while the spec definition of Lambda seem to be correct. 

```yaml
AWSTemplateFormatVersion: 2010-09-09
Resources:
  MyCognitoUserPool:
    Type: AWS::Cognito::UserPool
    Properties:
      UserPoolName: MyUserPool
      Policies:
        PasswordPolicy:
          MinimumLength: 8
      UsernameAttributes:
        - email
      Schema:
        - AttributeDataType: String
          Name: email
          Required: false
      UserPoolTags:
        - Key: Key1
          Value: Value1
        - Key: Key2
          Value: Value2
  ExampleLambda:
    Properties:
      Code:
        ZipFile: |
          exports.handler = async (event) => {
            console.log('Hello World');
            return 'Hello World';
          };
      Environment:
        Variables:
          - Key: A
            Value: B
          - C: d
      Handler: src/index.handler
      Runtime: nodejs18.x
      Role: arn:aws:iam::123456789012:role/MyRole
    Type: AWS::Lambda::Function

```


Deployment result
```
CREATE_FAILED                                                        AWS::Lambda::Function                                                ExampleLambda                                                        Properties validation failed for resource ExampleLambda with       
                                                                                                                                                                                                               message: #/Environment/Variables: expected type: JSONObject,       
                                                                                                                                                                                                               found: JSONArray                                                   
CREATE_FAILED                                                        AWS::Cognito::UserPool                                               MyCognitoUserPool                                                    Property validation failure: [Value of property {/UserPoolTags}    
                                                                                                                                                                                                               does not match type {Map}]
```

**Comment by kddejong** [comment]
Something seems off in how we are determining a valid map.

**Comment by kddejong** [comment]
Fixed and released in v0.77.0
